app/src/test/java/com/github/se/cyrcle/model/user/UserViewModelTest.kt

```diff
@@ -0,0 +1,90 @@
+package com.github.se.cyrcle.model.user
+
+import kotlinx.coroutines.runBlocking
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mock
+import org.mockito.MockitoAnnotations
+import org.mockito.kotlin.any
+import org.mockito.kotlin.eq
+import org.mockito.kotlin.verify
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class UserViewModelTest {
+
+  @Mock private lateinit var userViewModel: UserViewModel
+  @Mock private lateinit var userRepositoryFirestore: UserRepositoryFirestore
+
+  @Before
+  fun setUp() {
+    MockitoAnnotations.openMocks(this)
+    userViewModel = UserViewModel(userRepositoryFirestore)
+  }
+
+  @Test
+  fun addUserTest() {
+    userViewModel.addUser(UserTestInstances.user1)
+    // Check if the user was added to the repository
+    verify(userRepositoryFirestore).addUser(eq(UserTestInstances.user1), any(), any())
+  }
+
+  @Test
+  fun setCurrentUserTest() {
+    userViewModel.setCurrentUser(UserTestInstances.user1)
+
+    // Check if the user returned is the correct one
+    assert(userViewModel.currentUser.value == UserTestInstances.user1)
+  }
+
+  @Test
+  fun getUserByIdTest() = runBlocking {
```

Comment:
I am prety sure that the `runBlocking` is not usefull and can be removed. (To be confirmed by re-running the tests)

Reply:
You are right, thanks for the remark